internal/proxy/proxy.go

```diff
@@ -209,7 +216,11 @@ func NewClient(ctx context.Context, d cloudsql.Dialer, cmd *cobra.Command, conf
 			}
 		}
 
-		m := &socketMount{inst: inst.Name}
+		var opts []cloudsqlconn.DialOption
+		if inst.IAMAuthN {
+			opts = append(opts, cloudsqlconn.WithDialIAMAuthN(true))
```

Comment:
Do we error our somewhere if it's an unsupported engine? e.i. SQL Server? 

maybe that needs to be in the dialer

Reply:
https://github.com/GoogleCloudPlatform/cloud-sql-go-connector/issues/216